File: Python/problem_330.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
 
        sums = []
        missNumber = 1
        patches = 0
        i = 0
 
        while(missNumber <= n):
            if(i < len(nums) and missNumber >= nums[i]):
                missNumber += nums[i]
                i+=1          
            else:
                sums.append(missNumber)
                missNumber += missNumber
                patches += 1   
 
        return patches

File: Java/Problem330.java
class Solution {
    public int minPatches(int[] nums, int n) {
        List<Long> Sumas = new ArrayList<Long>();
        long Sum = 1;
        int Patches = 0;
        int i = 0;
        int size = nums.length;

        while(Sum <= n){
            if ((i < size) && (Sum >= nums[i])){
                Sum += nums[i];
                i++;
            }
            else{
                Sumas.add(Sum);
                Sum += Sum;
                Patches++;
            }
        }
        return Patches;
    }
}